woocommerce/woocommerce-android
# Move responsability of managing Toolbar aspects to the Fragments

### Description
This PR is an attempt to move the responsibility of managing the Toolbar to the fragments, this is useful now, and might be more useful if the team decides to delegate go with Composable Toolbars.
The PR introduces a property `activityAppBarStatus` to the `BaseFragment` class, that returns an `AppBarStatus` instance, and allows child fragments to override it, `MainActivity` uses the `AppBarStatus` instance to decide how to handle the Activity's AppBar and Toolbar, the different options allow handling: visibility, navigation icon, and shadow/divider.

@0nko @JorgeMucientes @hafizrahman since you were involved all in the discussions around the Toolbar, let me know what you think, and whether I should finish the PR or not (if this has any downsides that I'm missing).

### Testing instructions
1. Test the app and confirm the following screens properly hide the Activity Toolbar: "MoreFragment", "ProductImageViewerFragment" and "CouponDetailsFragment".
2. Test the following screens and confirm that the navigation icon is properly set to a `cross` icon: "ProductFilterListFragment" and "IssueRefundFragment"

### Images/gif
